### PR TITLE
RNTester - e2e cucumber new approach

### PR DESCRIPTION
**Pull request by adzironman**

### CODE REVIEW

File: packages/rn-tester-e2e/e2e-config.js [hunk 1]
```diff
@@ -4,21 +4,21 @@ const path = require('path');
...
 const android = {
-  platformName: 'Android',
-  platformVersion: '13.0',
-  deviceName: 'Pixel 6 API 33',
-  app: path.join(process.cwd(), '/apps/rn-tester.apk'),
-  automationName: 'UiAutomator2',
-  newCommandTimeout: 240,
+  'platformName': 'Android',
+  'appium:platformVersion': '12.0',
+  'appium:deviceName': 'Pixel_3a_API_33_arm64-v8a',
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/e2e-config.js]: I'm wondering if we shouldn't remove the value in capabilities. Shouldn't we leave blanks here (except path - I think it is useful here) ?

File: packages/rn-tester-e2e/e2e-config.js [hunk 2]
```diff
@@ -4,21 +4,21 @@ const path = require('path');
...
+  'appium:app': path.join(process.cwd(), '/apps/RNTester.apk'),
+  'appium:automationName': 'uiautomator2',
+  // 'appium:newCommandTimeout': 240,
 };
 
 const ios = {
   'platformName': 'iOS',
-  'appium:platformVersion': '16.0',
-  'appium:deviceName': 'iPhone 14 Pro Max',
+  'appium:platformVersion': '15.2',
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/e2e-config.js]: Same as above.

File: packages/rn-tester-e2e/README-describedAllFoldersAndFiles.md [hunk 3]
```diff
@@ -1,24 +1,21 @@
...
 ## helpers 🧑🏻‍🚒
-Utils file with generic, simple steps
+Utils file with generic, simple actions and methods
 
 ## runners 🏃🏽‍♀️
-Runner file which combines feature file and steps file. Runner file imports steps file and declares step functions in the same order as in the feature file (FUNCTION STEP LOGIC IS NOT IMPLEMENTED HERE!)
+Runner file which combines feature file and steps file. Runner file imports steps file and declares step functions in the same order as in the feature file
 
 ## screenObjects 📱
 Screen object files based on Page Object Pattern. One file defines all neccessary elements to interact with. These elements are defined as screen class variables, they are used by the steps file
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/README-describedAllFoldersAndFiles.md]: Small typo here: neccessary -> necessary.

File: packages/rn-tester-e2e/README-addTestAndExecute.md [hunk 4]
```diff
@@ -2,8 +2,8 @@
 
 1. Create a feature file `(rn-tester-e2e/test/features)` - GivenWhenThen Gherkin syntax
 2. **OPTIONAL -** Create a screen object or extend the existing one (depends on the test scope) - `rn-tester-e2e/test/screenObjects` - map screen elements for iOS and Android
-3. Create a step file `(rn-tester-e2e/test/steps)` - import used screen objects, create execution code of GivenWhenThen steps from feature file
-4. Create a runner file `(rn-tester-e2e/test/runners)` - import steps from step3 and create test scenarios
+3. **OPTIONAL -** Add another common step in `rn-tester-e2e/test/common_steps/common.steps.js`
+4. Create a runner file `(rn-tester-e2e/test/runners)` - import steps and screen objects from point 2 and 3. Create test scenarios
 5. Update `(rn-tester-e2e/e2e-config.js)` with proper cababilities of your emulator
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/README-addTestAndExecute.md]: Small typo here: cababilities -> capabilities.

File: packages/rn-tester-e2e/README-describedAllFoldersAndFiles.md [hunk 5]
```diff
@@ -27,4 +24,7 @@ Android and iOS emulator/physical device configuration, process.env.E2E_device g
 Global jest config setup - such as timeout, test runner path
 
 ## jest.setup.js
-Jest and wdio setup file
\ No newline at end of file
+Jest and wdio setup file
+
+## package.json
+all external dependecies and project parameters
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/README-describedAllFoldersAndFiles.md]: Small typo here: dependecies -> dependencies.

File: packages/rn-tester-e2e/test/runners/buttonComponentScreenRunner.js [hunk 6]
```diff
@@ -1,46 +1,86 @@
...
+import componentsScreen from '../screenObjects/components.screen.js';
+import buttonComponentScreen from '../screenObjects/buttonComponent.screen.js';
 
-Object.entries(steps).forEach(([name, exported]) => window[name] = exported);
+// Object.entries(steps).forEach(([name, exported]) => window[name] = exported);
 const feature = loadFeature('test/features/buttonComponentScreen.feature');
 
+//Methods used more than once in the feature
+const buttonComponentShouldBeDisplayed = (then) => {
+    then('Verify that the Button component is displayed', async () => {
```
Comment (elizabethberger) [inline packages/rn-tester-e2e/test/runners/buttonComponentScreenRunner.js:11]: Should we add "$" sign like on other methods?

**Review by elizabethberger — CHANGES_REQUESTED** [review]
Most of my comments are small things - we can discuss any.

File: packages/rn-tester-e2e/test/runners/buttonComponentScreenRunner.js (hunk 6 above)
Comment (adzironman) [reply packages/rn-tester-e2e/test/runners/buttonComponentScreenRunner.js:11]: I would leave it as it is for now. If we will be checking another component, then we can refactor that method

**Pull request merged**